Content.Shared/Turnstile/Systems/TurnstileSystem.cs

```diff
@@ -0,0 +1,180 @@
+    }
+
+
+    private void HandleCollide(Entity<TurnstileComponent> ent, ref StartCollideEvent args)
+    {
+        // The colliding entity needs to be a mob.
+        if (!HasComp<MobStateComponent>(args.OtherEntity))
+            return;
+
+        if (!TryComp<TransformComponent>(ent, out var xform))
+            return;
+
+        // Check the contact normal against our direction.
+        // For simplicity, we always want a mob to pass from the "back" to the "front" of the turnstile.
+        // That allows unanchored turnstiles to be dragged and rotated as needed, and will admit passage in the
+        // direction that they are pulled in.
+        var facingDirection = xform.LocalRotation.GetDir();
+        var directionOfContact = GetDirectionOfContact(xform, args.OtherEntity);
+
+        if (ent.Comp.State == TurnstileState.Rotating || facingDirection != directionOfContact)
+        {
+            // Reject the entity, play sound
+            _audio.PlayPredicted(ent.Comp.BumpSound, ent, args.OtherEntity, AudioParams.Default.WithVolume(-3));
+            return;
+        }
+
+        // Admit the entity.
+        var comp = EnsureComp<PreventCollideComponent>(ent);
+        comp.Uid = args.OtherEntity;
+
+        // Play sound of turning
+        _audio.PlayPredicted(ent.Comp.TurnSound, ent, args.OtherEntity, AudioParams.Default.WithVolume(-3));
+
+        // We have to set collidable to false for one frame, otherwise the client does not
+        // predict the removal of the contacts properly, and instead thinks the mob is colliding when it isn't.
+        SetCollidable((ent, ent.Comp, null), false);
+        SetState((ent, ent.Comp), TurnstileState.Rotating);
+    }
+
+    /// <summary>
+    ///     Turnstiles stop rotating once their admitted entity has passed through.
+    /// </summary>
+    private void StopRotatingIfEntityPassed(Entity<TurnstileComponent> ent)
+    {
+        if (ent.Comp.State != TurnstileState.Rotating)
+            return;
+
+        // Check and see if we're still colliding with the admitted entity.
+        var stillCollidingWithAdmitted = GetIsCollidingWithAdmitted(ent);
+        if (stillCollidingWithAdmitted)
+            return;
+
+        SetState(ent, TurnstileState.Idle);
+        RemComp<PreventCollideComponent>(ent);
+    }
+
+    private bool GetIsCollidingWithAdmitted(EntityUid uid, TurnstileComponent? turnstile = null, TransformComponent? xform = null, PreventCollideComponent? preventCollide = null)
+    {
+        if (!Resolve(uid, ref turnstile) || !Resolve(uid, ref xform) || !Resolve(uid, ref preventCollide))
+            return false;
+
+        if (!TryComp<TransformComponent>(preventCollide.Uid, out var otherxform))
+            return false;
+
+        var turnstileAABB = _entityLookupSystem.GetAABBNoContainer(uid, xform.LocalPosition, xform.LocalRotation);
+        var otherAABB = _entityLookupSystem.GetAABBNoContainer(uid, otherxform.LocalPosition, otherxform.LocalRotation);
+
```

Comment:
I don't know where you got this from but I left this TODO above the content usages:

`// TODO: This shouldn't be calculating based on world AABBs.`

Reply:
OK, now uses a local bounds solution similar to the conveyor belt, using the mapgrid tile bounds and a simple bounds-to-center check for the target entities.